localstack/services/dynamodbstreams/dynamodbstreams_api.py

```diff
@@ -64,6 +65,20 @@ def get_stream_for_table(table_arn):
     return region.ddb_streams.get(table_name)
 
 
+def check_stream_ready(kinesis, stream_name):
+    status = kinesis.describe_stream(StreamName=stream_name)["StreamDescription"]["StreamStatus"]
+    if status not in ("ACTIVE", "UPDATING"):
+
+        def check_stream_active():
```

Comment:
Personally, I think that using a nested function here with `retry` is unnecessarily complicated. You could accomplish the same thing with a `for` loop.

Reply:
I added this part, because I had a problem during testing. But it seems to be gone now, at least I cannot reproduce it anymore - will remove this therefore :)